test/llmg.ts

```diff
@@ -0,0 +1,382 @@
+  // We need to figure out which tenant, context, and feature id to use
+  // Maybe this is something that will be given to us once the client registration completes???
+  'x-sfdc-core-tenant-id': 'core/prod1/00DDu0000008cuqMAA',
+  'x-sfdc-app-context': 'EinsteinGPT',
+  'x-client-feature-id': 'EinsteinDocsAnswers',
+});
+
+const createRequestBody = (
+  model: string,
+  tools: InvocableTool[],
+  messages: Array<{ role: string; content: string }>
+): string =>
+  JSON.stringify({
+    model,
+    tools,
+    tool_config: {
+      mode: 'auto',
+    },
+    messages,
+    generation_settings: {
+      max_tokens: 500,
+      temperature: 0.5,
+      parameters: {},
+    },
+  });
+
+const makeSingleGatewayRequest = async (
+  model: string,
+  tools: InvocableTool[],
+  messages: Array<{ role: string; content: string }>
+): Promise<GatewayResponse> => {
+  const response = await fetch(
+    'https://bot-svc-llm.sfproxy.einsteintest1.test1-uswest2.aws.sfdc.cl/v1.0/chat/generations',
+    {
+      method: 'POST',
+      headers: createRequestHeaders(),
+      body: createRequestBody(model, tools, messages),
+    }
+  );
+
+  if (!response.ok) {
+    throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+  }
+
+  const responseData = await response.json();
+  return responseData as GatewayResponse;
+};
+
+/**
+ * Makes requests to the LLM Gateway API for multiple prompts using the specified model and tools.
+ *
+ * @param {string[]} prompts - Array of prompts to send to the API
+ * @param {string} model - The model identifier to use for generation (e.g., 'llmgateway__AzureOpenAIGPT4Omni')
+ * @param {InvocableTool[]} tools - Array of tools that can be invoked by the model
+ * @returns {Promise<{model: string, messages: Array<{role: string, content: string}>, responses: GatewayResponse[]}>} Object containing the model used, conversation messages, and API responses
+ * @throws {Error} If any API request fails or returns an error
+ *
+ * @see {@link https://git.soma.salesforce.com/pages/tech-enablement/einstein/docs/gateway/get-started/#make-your-first-gateway-request} Make Your First Gateway Request Documentation
+ * @see {@link https://git.soma.salesforce.com/pages/tech-enablement/einstein/docs/gateway/models-and-providers/} Models and Providers Documentation
+ * @see {@link https://git.soma.salesforce.com/pages/tech-enablement/einstein/docs/gateway/apis/rest/#operation/chatMessages} REST API Documentation
+ * @see {@link https://git.soma.salesforce.com/pages/tech-enablement/einstein/docs/gateway/function-calling/} Function Calling Documentation
+ * @see {@link https://git.soma.salesforce.com/pages/tech-enablement/einstein/docs/gateway/get-started/auth/#api-key-limitations} API Key Limitations Documentation
+ */
+const makeGatewayRequests = async (
+  prompts: string[],
+  model: string,
+  tools: InvocableTool[]
+): Promise<{ model: string; messages: Array<{ role: string; content: string }>; responses: GatewayResponse[] }> => {
```

Comment:
should model values be verified against an allow list first to avoid making calls with bad (likely accidental) input?